src/library/scala/collection/immutable/ChampHashMap.scala

```diff
@@ -575,10 +575,12 @@ private final class HashCollisionMapNode[K, +V](val originalHash: Int, val hash:
   import Node._
```

Comment:
is there any good reason why HashCollisionMapNode used a Vector rather than an Array?
an Array is faster, smaller etc ( but less functionally pure, and internally mutable)

Reply:
Vector offers the opportunity for structural sharing, whereas if you use a contiguous array you always need to copy the whole structure each time a colliding element is added or removed. Not sure how much this helps in practice, though.

Reply:
yes - that was my thought. It would seem that an Array[Any] with alternating key/value would be smaller and faster. 

We generally don't expect many values to collide, but `Vector[Tuple2(K,V)]` seems heavy on memory and unlikely to share much except if we have >32 elements that collide and we remove the last one (and you don't have control of the ordering)

Also it would be more similar to the structure in `BitmapIndexedMapNode`.